### PR TITLE
Update config.yml to new orb

### PR DESCRIPTION
**Pull request by allenhsieh**
Update config.yml to new orb and made adjacent pom.xml changes

This pull request makes the following changes:
* Update config.yml to new orb and made adjacent pom.xml changes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,49 +2,30 @@ version: 2.1
 
 orbs:
   github-maven-deploy: github-maven-deploy/github-maven-deploy@1.0.5
```
Comment (DarthHater) [inline .circleci/config.yml]: This is still needed if you use ours?

Comment (allenhsieh) [reply .circleci/config.yml]: Yeah! from my understand. It's in the R repo i'm using as my main reference: https://github.com/sonatype-nexus-community/nexus-repository-r/blob/292230836d77aa8445e70ce81793fcdda0395de1/.circleci/config.yml#L4

Comment (bhamail) [reply .circleci/config.yml]: @allenhsieh is correct-we still use Benny’s build-and-test job

File: pom.xml [hunk 2]
```diff
@@ -73,24 +78,44 @@
...
-      <activation>
-        <property>
-          <name>performRelease</name>
-          <value>true</value>
-        </property>
-      </activation>
       <build>
         <plugins>
           <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
```
Comment (bhamail) [inline pom.xml]: Minor: you actually don’t need to provide a groupId when the Id is o.a.m.p. Therefore, “convention over configuration” dictates it should be left out.

Comment (allenhsieh) [reply pom.xml]: Thank you! Should I go ahead and move it from the R format as well?         https://github.com/sonatype-nexus-community/nexus-repository-r/blob/292230836d77aa8445e70ce81793fcdda0395de1/pom.xml#L93

File: .circleci/config.yml (hunk 1 above)
Comment (DarthHater) [reply .circleci/config.yml]: For some wild reason I thought the new one `extend`ed the previous one, good to know!

**Pull request merged**